lib/direncrypt/inventory.py

```diff
@@ -55,38 +55,103 @@ def read_parameters(self, params_only=False):
+        request = """SELECT unencrypted_file, encrypted_file, public_id,
+                     is_link, target FROM register WHERE is_link=0 and
+                     encrypted_file <>''"""
+        for row in self.cursor.execute(request):
+            rows[row[0]] = {
+                'unencrypted_file': row[0],
+                'encrypted_file':   row[1],
+                'public_id':        row[2],
+                'is_link':          row[3],
+                'target':           row[4]
+            }
+        return rows
+    
+    def read_registered_links(self):
+        """Get information on all registered links.
+        
+        Returns a dict with unencrypted filename for keys, having
+        a dict of unencrypted file, encrypted file, public id,
+        is_link and target as value.
+        """
+        rows = {}
+        request = """SELECT unencrypted_file, encrypted_file, public_id, 
+                     is_link, target FROM register WHERE is_link=1"""
+        for row in self.cursor.execute(request):
+            rows[row[0]] = {
+                'unencrypted_file': row[0],
+                'encrypted_file':   row[1],
+                'public_id':        row[2],
+                'is_link':          row[3],
+                'target':           row[4]
+            }
+        return rows
+    
+    def read_registered_dirs(self):
+        """Get information on all registered empty directories.
+        
+        Returns a dict with unencrypted filename for keys, having
+        a dict of unencrypted file, encrypted file, public id,
+        is_link and target as value.
+        """
+        rows = {}
+        request = """SELECT unencrypted_file, encrypted_file, public_id,
+                     is_link, target FROM register WHERE is_link=0 and
+                     encrypted_file=''"""
+        for row in self.cursor.execute(request):
+            rows[row[0]] = {
+                'unencrypted_file': row[0],
+                'encrypted_file':   row[1],
+                'public_id':        row[2],
+                'is_link':          row[3],
+                'target':           row[4]
             }
         return rows
     
     def read_line_from_register(self, plainfile):
         """Get encrypted filename from unencrypted filename in register"""
         result = {}
-        for row in self.cursor.execute('''
-                SELECT encrypted_file FROM register WHERE unencrypted_file = ? ''',(plainfile,)):
+        request = "SELECT encrypted_file FROM register WHERE unencrypted_file = ?"
+        for row in self.cursor.execute(request, (plainfile,)):
             result[plainfile] = {'encrypted_file':   row[0]}
         return result[plainfile]['encrypted_file']
 
-    def register(self, plain_path, enc_path, public_id):
+    def register(self, plain_path, enc_path, public_id, is_link, target):
         """Register input and output filenames into a database."""
+        is_link_int = 0 if is_link==False else 1
```

Comment:
Please drop this in favour of `int(is_link)`.

Reply:
done